File: creational/singleton/singleton.java
public class SingletonPattern {
    public static class Simba {
        private static Simba uniqueInstance;

        public static synchronized Simba getInstance() {
            if (uniqueInstance == null) {
                uniqueInstance = new Simba();
            }
            return uniqueInstance;
        }

        private String name = "Simba";

        private Simba() { }

        public void makeSound() {
            String sound = String.format("I'm unique %s! Meow!", name);
            System.out.println(sound);
        }
    }

    public static void main(String[] args) {
        Simba simba = Simba.getInstance();
        simba.makeSound();
    }
}

File: creational/singleton/singleton.cs
class SingletonPattern
{
    class Simba
    {
        private static Simba uniqueInstance { get; set; }

        public static Simba GetInstance()
        {
            if (uniqueInstance == null)
            {
                uniqueInstance = new Simba();
            }
            return uniqueInstance;
        }

        private string name { get; } = "Simba";

        private Simba() { }

        public void MakeSound()
        {
            System.Console.WriteLine("I'm unique {0}! Meow!", name);
        }
    }

    static void Main(string[] args)
    {
        Simba simba = Simba.GetInstance();
        simba.MakeSound();
    }
}
